### PR TITLE
Adding params. to create_auto_ml_forecasting_training_job in AutoMl hook

### PR DESCRIPTION
**Pull request by vinay2242g**
Added window_stride_length & window_max_count

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by vinay2242g** [comment]
Hi,

I am really really new to open source community. I may not be fully aware lot of things here.

Please help me grow here.

Thanks,

File: airflow/providers/google/cloud/hooks/vertex_ai/auto_ml.py
Comment (shahar1) [inline airflow/providers/google/cloud/hooks/vertex_ai/auto_ml.py]:
```suggestion
        window_stride_length: int | None = None,
        window_max_count: int | None = None,
```

File: airflow/providers/google/cloud/hooks/vertex_ai/auto_ml.py
Comment (shahar1) [inline airflow/providers/google/cloud/hooks/vertex_ai/auto_ml.py]:
```suggestion
            window_max_count=window_max_count,
```

File: airflow/providers/google/cloud/hooks/vertex_ai/auto_ml.py
Comment (shahar1) [inline airflow/providers/google/cloud/hooks/vertex_ai/auto_ml.py]:
```suggestion
        :param window_stride_length: Optional. Step length used to generate input examples. Every 
        ``window_stride_length`` rows will be used to generate a sliding window.
        :param window_max_count: Optional. Number of rows that should be used to generate input examples. If the 
        total row count is larger than this number, the input data will be randomly sampled to hit the count.
```

File: airflow/providers/google/cloud/operators/vertex_ai/auto_ml.py
Comment (shahar1) [inline airflow/providers/google/cloud/operators/vertex_ai/auto_ml.py]:
```suggestion
        window_stride_length: int | None = None,
        window_max_count: int | None = None,
```

File: airflow/providers/google/cloud/operators/vertex_ai/auto_ml.py
Comment (shahar1) [inline airflow/providers/google/cloud/operators/vertex_ai/auto_ml.py]:
```suggestion
        self.window_stride_length=window_stride_length,
        self.window_max_count=window_max_count,
```

File: airflow/providers/google/cloud/operators/vertex_ai/auto_ml.py
Comment (shahar1) [inline airflow/providers/google/cloud/operators/vertex_ai/auto_ml.py]:
```suggestion
            window_stride_length=self.window_stride_length,
            window_max_count=self.window_max_count,
```

**Review by shahar1 — CHANGES_REQUESTED** [review]
Welcome to Apache Airflow, and thank you for your first contribution!
You may want to read the [contributing docs](https://github.com/apache/airflow/blob/main/contributing-docs/README.rst) to familiarize yourself with what you need to contribute here.
As for this specific PR, adding these parameters is a great idea - but there are a few things that you need to take care of before we'll be able to merge it:
1. Fix docstrings and parameter assignment—To help you, I created suggestions that you can commit via GitHub UI or manually.
2. Update the unit test in `tests/providers/google/cloud/operators/test_vertex_ai.py` in lines 1280-1343). It's a matter of adding the parameters to the operator call and the hook assertions to ensure it's passed correctly. If you're not familiar with the concept of unit tests, I suggest that you read some material about it (for example, in [Real Python](https://realpython.com/python-testing/)), as well as about the testing framework that we use, which is called [pytest](https://docs.pytest.org/en/8.2.x/).
3. Change the PR's title to something more indicative (for example, `Adding params. to create_auto_ml_forecasting_training_job in AutoMl hook`).

If you're not sure of what or how to do the above, feel free to ask :)
Good luck!

**Comment by vinay2242g** [comment]
Thank you Shahar,


I have tried to implement things that you have mentioned.

Could you please let me know if they look good.

Thanks,

**Review by dirrao — APPROVED** [review]
LGTM. Congratulations on your first PR.

**Review by shahar1 — APPROVED** [review]
Looks good!
The tests need some improvement, but it could wait for other time IMO

**Comment by vinay2242g** [comment]
Thank you Shahar,

Could you please let me if there any action item on my side?


Thanks,
Siddesh

**Review by vinay2242g — COMMENTED** [review]
removed spaces

**Review by vinay2242g — COMMENTED** [review]
removed spaces

**Review by shahar1 — APPROVED** [review]

**Comment by shahar1** [comment]
> Thank you Shahar,
> 
> Could you please let me if there any action item on my side?
> 
> Thanks, Siddesh

It looks good to me, but we need approval from a committer so it can be merged.
@eladkal we'd be happy for your help :)

File: airflow/providers/google/cloud/hooks/vertex_ai/auto_ml.py
Comment (josh-fell) [inline airflow/providers/google/cloud/hooks/vertex_ai/auto_ml.py]:
```suggestion
        :param window_stride_length: Optional. Step length used to generate input examples. Every 
            ``window_stride_length`` rows will be used to generate a sliding window.
        :param window_max_count: Optional. Number of rows that should be used to generate input examples. If the 
            total row count is larger than this number, the input data will be randomly sampled to hit the count.
```
Small nit on [formatting the docstring params](https://sphinx-rtd-tutorial.readthedocs.io/en/latest/docstrings.html#an-example-class-with-docstrings) in the Sphinx RTD format for the [Python API docs](https://airflow.apache.org/docs/apache-airflow-providers-google/stable/_api/airflow/providers/google/cloud/hooks/vertex_ai/auto_ml/index.html#airflow.providers.google.cloud.hooks.vertex_ai.auto_ml.AutoMLHook.create_auto_ml_forecasting_training_job).

**Review by josh-fell — APPROVED** [review]
Small nit on docstring format, but LGTM!

File: airflow/providers/google/cloud/operators/vertex_ai/auto_ml.py
Comment (Taragolis) [inline airflow/providers/google/cloud/operators/vertex_ai/auto_ml.py]:
```suggestion
        self.window_stride_length = window_stride_length
        self.window_max_count = window_max_count
```

In this case comma in the end transform into the tuples `(window_stride_length, )` and `(window_max_count, )` 
I guess it added accidentally

Comment (Taragolis) [reply airflow/providers/google/cloud/operators/vertex_ai/auto_ml.py]: And also suggested changes will fix the tests

**Comment by josh-fell** [comment]
Congrats @vinay2242g 🎉

**Pull request merged**
